### PR TITLE
Refactor gatekeeper lambda to retrieve Github repo id

### PR DESCRIPTION
**Pull request by emilymcafee**
From @rclark: 

> In stork's .setupHook function, there are two API calls made:
>
>   1. Find out the repository ID number given a repository name
>   2. List the repo as one that the stork github app should pay attention to
>
> The first API call can be accomplished using a user's personal access token, but the second has to be an org admin. Right now in that function, the user passes in their token, which is used in the first request, but then they invoke a lambda function to perform the second call.
> 
> That's silly. Let's just refactor the Lambda function so that the user provides the repository name, and then the lambda function makes both requests.

This PR does that by bringing the Github repo ID lookup into the lambda function that has an org admin Github access token set in its environment. This allows us to remove the `token` input from the .setupHook function.

TODO:
  - [ ] Figure out if there are other docs or places like bin/hook.js that need to be adjusted with this change.
  - [ ] Docs, changelog, etc
  - [ ] Deploy new code to running Lambda function, run some tests that way.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -18,44 +17,22 @@ const setupHook = (options) => {
...
+        .find((output) => output.OutputKey === 'GithubAppInstallationId')
+        .OutputValue);
+      const FunctionName = outputs
+        .find((output) => output.OutputKey === 'GatekeeperLambda')
+        .OutputValue;
+
+      return lambda.invoke({
+        FunctionName,
+        InvocationType: 'RequestResponse',
+        Payload: JSON.stringify({ repo: options.repo, org: options.org, installationId: installationId })
```
Comment (emilymcafee) [inline index.js]: @rclark this `installationId` parameter is carried over from before. However, poking around the lambda function code I can't actually find where this is used -- that function seems to have the installation ID already set as `process.env.GITHUB_APP_INSTALLATION_ID`. Does that make this `installationId: installationId` totally extraneous, or am I missing something?

Comment (rclark) [reply index.js]: This was probably me trying to plan for a single gatekeeper lambda that could add **any** repo to **any** github app installation. I think we want to end up with such a function, but maybe don't need it as part of stork.

File: lambda.js [hunk 2]
```diff
@@ -585,11 +585,24 @@ stork.forwarder = (event, context, callback) => {
...
     .then(() => {
+      const query = { access_token: process.env.GITHUB_ACCESS_TOKEN };
+      const config = {
+        json: true,
+        headers: {
+          'User-Agent': 'github.com/mapbox/stork',
+          'Content-Type': 'application/json'
+        }
+      };
+      const uri = `https://api.github.com/repos/${event.repo}/${event.org}`;
```
Comment (rclark) [inline lambda.js]: This looks backwards. Shouldn't it be `/repos/org/repo`?

**Comment by arunasank** [comment]
After https://github.com/mapbox/mbxcli/pull/811 and https://github.com/mapbox/github-apps/pull/8, we don't need the setup hook/gatekeeper function anymore, so deleted these files and associated tests. The environment variables associated with the app, the installation ID, the private key etc are still to be retained though, for the trigger/status lambda functions.

**Comment by arunasank** [comment]
@rclark can you take a second look?

**Comment by rclark** [comment]
I'm going to close this PR for now -- we have sidestepped the issues we were facing internally.

**Pull request closed without merging**